### PR TITLE
Add and remove todolist

### PR DESCRIPTION
**Pull request by Yemisirach**
### Project requirements
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi @Yemisirach, 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- GitHub flow is used correctly. ✔
- All linters checks are ok. ✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Yemisirach** [comment]
- I do default make sure that the todos array has the property completed set to false and the property index set to the value of the new array length. And if an item is deleted that should update all remaining items' indexes.
- fix delete and edit

File: src/index.js [hunk 1]
```diff
@@ -1,50 +1,114 @@
...
+
+const editTask = document.querySelector('.editTask');
+
+editTask.addEventListener('click', (taskId, textName) => {
+  editId = taskId;
+  isEditTask = true;
+  todoInput.value = textName;
+  todoInput.focus();
+  todoInput.classList.add('active');
+});
```
Comment (Tadesse-Alemayehu) [inline src/index.js:105]: - [ ] Dear @Yemisirach, as the previous reviewer mentioned, The editing functionality is currently not working. It is one of the requirements that a user be able to edit a task that they have added. Please correct this.

File: src/index.js [hunk 2]
```diff
@@ -1,50 +1,114 @@
...
+
+const deleteTask = document.querySelector('.deleteTask');
+
+deleteTask.addEventListener('click', (deleteId, filter) => {
+  isEditTask = false;
+  todos.splice(deleteId, 1);
+  localStorage.setItem('todo-list', JSON.stringify(todos));
+  showTodo(filter);
+});
```
Comment (Tadesse-Alemayehu) [inline src/index.js:114]: - [ ] Dear @Yemisirach, as the previous reviewer mentioned, the ability to delete a task is not functional. Please address this as well.

File: src/index.js [hunk 3]
```diff
@@ -1,50 +1,114 @@
...
-
-    ToDoTask.append(listItem);
-  });
+const todoInput = document.querySelector('.todo-input input');
+const taskBox = document.querySelector('.task-box');
+const clearAll = document.querySelector('.clear');
+const filters = document.querySelectorAll('.filters span');
+let editId;
+let isEditTask = false;
+let todos = JSON.parse(localStorage.getItem('todo-list'));
```
Comment (Tadesse-Alemayehu) [inline src/index.js:9]: - [ ] Dear @Yemisirach, as the previous reviewer mentioned, the property index is required to be set to the value of the new array length. And if an item is deleted that should update all remaining items' indexes. currently all your task are having index of `0.`

**Review by Tadesse-Alemayehu — CHANGES_REQUESTED** [review]
### Hi @Yemisirach 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [ ] Good job making most of the changes requested by the previous reviewer. 🙏
 - [ ] Good job overall following the template design 💯

 ## Required Changes ♻️
- [ ] Dear @Yemisirach, we need to have a functioning `add`, `edit` and `remove` operations in our Todo list. 

https://user-images.githubusercontent.com/69077061/215290431-4e9c9c3d-e60e-4da3-bb2e-638dfd62128c.mp4


_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (Yemisirach) [reply src/index.js:105]: hello tadesse  can you help me on that?

File: src/index.js (hunk 3 above)
Comment (Yemisirach) [reply src/index.js:9]: I do this part. you can check on local storage.

File: src/index.js (hunk 1 above)
Comment (Tadesse-Alemayehu) [reply src/index.js:105]: Sorry for reaching back this late @Yemisirach.

**Comment by Yemisirach** [comment]
- I do default make sure that the todos array has the property completed set to false and the property index set to the value of the new array length. And if an item is deleted that should update all remaining items' indexes.
- fix delete and edit

**Comment by huseyinbiyik** [comment]
Hi @Yemisirach,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

Alt Text

## Invalid Code Review Request

⚠️ You have submitted a project, that has no Readme file and no instruction on how to run the project. Currently, after cloning the project, `npm install` and `npm run start` commands are giving error and live server is not starting. Kindly fix the issue before submitting for the review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Yemisirach** [comment]
update README File

**Review by GedeonTS — APPROVED** [review]
# STATUS: APPROVED :tada: :clap: :champagne: 
Hi @Yemisirach ,

You are amazing :smiley: :clap: !  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

 

## Highlights :clap: :tophat: 
- Add, Edit and Remove are   working :+1: 
- The indexes are getting updated when the user deletes :+1: 
## Optional suggestions :small_airplane: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**